### PR TITLE
stm32: interrupt_controller: Access line-range prop elements at build time

### LINKED ISSUE #57263: llvm/arm-clang fails to build STM32 platforms
when building with arm-clang or llvm for arm we get the following error:

```
/home/galak/git/zephyr/drivers/interrupt_controller/intc_exti_stm32.c:231:9: error: initializer element is not a compile-time constant
```

To reproduce, you can grab the ARM LLVM prebuilt here:

https://github.com/ARM-software/LLVM-embedded-toolchain-for-Arm/releases

```
export ZEPHYR_TOOLCHAIN_VARIANT=llvm
export LLVM_TOOLCHAIN_PATH=<PATH_TO_LLVM>
```

```
west build -p -b nucleo_f767zi  -T tests/kernel/common/kernel.common
```

### PR DESCRIPTION
**Pull request by erwango**
Fixes #57263

Involves the re-introduction of UTIL_INC and new UTIL_X2 method.
For coherency UTIL_DEC is added as well.

EDIT: Fixes regression cause by https://github.com/zephyrproject-rtos/zephyr/pull/58421/commits/ba5e50c20006677788ac3a839227c09a10eaec88 (breaking communication on H5/U5/L5 targets)

### CODE REVIEW

File: include/zephyr/sys/util_internal.h [hunk 1]
```diff
@@ -407,4 +407,785 @@
...
+#define UTIL_DEC_248 247
+#define UTIL_DEC_249 248
+#define UTIL_DEC_250 249
+#define UTIL_DEC_251 250
+#define UTIL_DEC_252 251
+#define UTIL_DEC_253 252
+#define UTIL_DEC_254 253
+#define UTIL_DEC_255 254
+
+#define UTIL_x2_0 0
```
Comment (mbolivar-nordic) [inline include/zephyr/sys/util_internal.h:934]: Nonblocking, but using a lowercase "x" here feels wrong to me. Shouldn't API macros be capitalized?

File: include/zephyr/sys/util_macro.h [hunk 2]
```diff
@@ -366,6 +366,22 @@ extern "C" {
...
+ * Similarly, UTIL_DEC(x) is (x-1) as an integer literal.
+ */
+#define UTIL_INC(x) UTIL_PRIMITIVE_CAT(UTIL_INC_, x)
+#define UTIL_DEC(x) UTIL_PRIMITIVE_CAT(UTIL_DEC_, x)
+
+/**
+ * @brief UTIL_x2(y) for an integer literal y from 0 to 255 expands to an
+ * integer literal whose value is 2y.
+ */
+#define UTIL_x2(y) UTIL_PRIMITIVE_CAT(UTIL_x2_, y)
```
Comment (mbolivar-nordic) [inline include/zephyr/sys/util_macro.h:382]: This one in particular.

**Review by mbolivar-nordic — DISMISSED** [review]

**Comment by erwango** [comment]
@mbolivar-nordic Alternative solution poped up which doesn't involves `UTIL_` macros.
Hence, aside from the readability (which could be worked on), difference between 2 PRs is the (re-)introduction of `UTIL_` macros.
So question is: do we think these macros are a good thing to add ?

**Comment by erwango** [comment]
> Hence, aside from the readability (which could be worked on), difference between 2 PRs is the (re-)introduction of `UTIL_` macros.
> So question is: do we think these macros are a good thing to add ?

[This comment](https://github.com/zephyrproject-rtos/zephyr/pull/58755#issuecomment-1576825754) tends to demonstrate interest in these macros.

**Review by gautierg-st — APPROVED** [review]

**Comment by erwango** [comment]
@semihalf-duda-patryk Would you mind providing a review? This would help getting this merged in RC2

**Review by FRASTM — APPROVED** [review]

**Review by duda-patryk — APPROVED** [review]

**Pull request merged**